### ISSUE TITLE
Dash-To-Dock corner bug

### ISSUE THREAD

**Issue by TheAifam5** [adapta-project/adapta-gtk-theme#256]
**Reporter info**

 ```
 * Distribution - ArchLinux
 * Gtk+ 3.0 version - 3.22.1
 * Desktop environment - Gnome Desktop
 * Related issue on dash-to-dock: https://github.com/micheleg/dash-to-dock/issues/388
 ```

**Adapta version (if you knew)**
Git - cf7267016c68062f2ebfcce5f51bbcf9a5435dee

**Related Application and/or Shell theming (or name of widget)**
Dash-to-dock Gnome extension

**Actual issue**
The corners contains rectangles and lines.

**Steps to reproduce (if you knew)**
Install Gnome Desktop
Install theme
Install Dash-To-Dock (git version)

**Expected behaviour**
Previously was without black rectangles below rounded corners.

**Comment by tista500** [comment]
"Shrink the dash" option overrides `border-radius`, so how we protect the background-area?

**Comment by tista500** [comment]
That option breaks other themes as well, so please stop using that option ( it seems to draw unwanted borders around the dock, too).

Refector styling: https://github.com/adapta-project/adapta-gtk-theme/commit/58887918bea1d24c1996491790786ac5f5eab899

This commit brings much smaller dock area than that option was enabled.
Closed.

**Comment by TheAifam5** [comment]
Ok, i will stop using it.
The best idea will be contact with author of Dash-to-Dock.

**Comment by MS7WN7D-V3N0M** [comment]
I have the same issue mate.
I cannot stop using it, nor stop using Adapta I just love both of them !

**Comment by micheleg** [comment]
It seems to be working fine to me now. Has the issue been fixed? Also, I'm not sure if #306 is related.

**Comment by tista500** [comment]
@micheleg,
> I'm not sure if #306 is related.

I don't think that typo was related to this issue.
And at least version d18f2d6 has changed nothing.

As I read your stylesheet.css, `.shrink` style-class causes these things in Adapta:
* 9px radius break the design.
* 1px solid border around the dash breaks the design.
* `.shrink` style-class pulls `@at-root .app-well-app .overview-icon` styling to create `.dash-item-container > StWidget`. This behaviour overrides our specific `@at-root .dash-item-container > StWidget` spacing properties.
* Basically we have just 1px padding in `#dash`. Reducing 1px to 0 does not mean "shrink".

Especially the combo `border-radius` and `border-width` seem to break the design of other themes as well, Arc, Flat-Plat (and some more?). So all I want is notifying users by adding "This option breaks Adapta theme" in the comment of your configuration window.

Regards.

**Comment by micheleg** [comment]
So there are two issue:
1. my shrinking workaround is pointless - actually counterproductive - with your theme. For this I can only warn the user as you suggest
2. An ugly corner appear with your theme and the shrink option. It seems to be due to the setting of the property "box-shadow: none" on the dash:

https://github.com/adapta-project/adapta-gtk-theme/blob/master/shell/sass/gnome-shell/_common.scss#L1852

https://github.com/adapta-project/adapta-gtk-theme/blob/master/shell/sass/gnome-shell/_common.scss#L1895

I don't understand why (maybe a bug?), but what the reason for this property to be set?

**Comment by tista500** [comment]
> I don't understand why (maybe a bug?), but what the reason for this property to be set?

This was for my future works to implement raised materials.

**Comment by micheleg** [comment]
OK. I don't want to interfere with your plans, just pointing out a possible issue/fix.

**Comment by tista500** [comment]
Ah I now confirmed those `box-shadow` properties remained the background areas behind `.shrink` class. So I will comment them until changes has come.

Regards.

**Comment by tista500** [comment]
Committed: https://github.com/adapta-project/adapta-gtk-theme/commit/6d63abd03077d93243caeb8ce06ee94f82f81cf3
